base/compiler/optimize.jl

```diff
@@ -509,7 +513,7 @@ function finish(interp::AbstractInterpreter, opt::OptimizationState,
                     cost_threshold += 4*default
                 end
             end
-            src.inlineable = inline_worthy(ir, params, union_penalties, cost_threshold)
+            src.inlining_cost = UInt16(inline_worthy(ir, params, union_penalties, cost_threshold))
```

Comment:
This conversion can throw. You will need to make sure that `inline_worthy` (maybe best to rename to `inline_cost`) saturates at `typemax(UInt16)`.

(Does `inline_worthy` already return the inlining cost?)